### PR TITLE
Issue 322 - Add Postgres and Nginx Libraries

### PR DESCRIPTION
**Pull request by PatrickMessierGoC**
Also included:
- Fix for issue-933 - Fail to create database and username containing hyphen character
- Fix for issue-934 - Add function to reset user password in mysql library
- Refactoring of logger.sh

### CODE REVIEW

**Review by cedricguindon — APPROVED** [review]

**Comment by PatrickMessierGoC** [comment]
Pour @cedricguindon 
"So when doing IFS= read -r line, what you are doing is setting IFS to an empty string (no character will be used to split, therefore no splitting will occur) so that read will read the entire line and see it as one word that will be assigned to the line variable. The changes to IFS only affect that statement, so that any following commands won't be affected by the change."
Ref.: https://unix.stackexchange.com/questions/209123/understanding-ifs-read-r-line

File: ubuntu-18.04/mysql.sh [hunk 1]
```diff
@@ -13,6 +13,56 @@ readonly MYSQL_USER_OPTIONS_FILE_PATH="${HOME}/.my.cnf"
...
+  local database_server_new_credentials_password="$5"
+
+  # Variables
+  local database_server_new_credentials_database="${database_server_new_credentials_username}"
+
+  mysql::create_user_options_file \
+    "${database_server_admin_username}" \
+    "${database_server_admin_password}" \
+    "${database_server_fqdn}" \
+    "5432" \
```
Comment (mathieupeterson) [inline ubuntu-18.04/mysql.sh]: 3306? mysql

File: ubuntu-18.04/mysql.sh [hunk 2]
```diff
@@ -13,6 +13,56 @@ readonly MYSQL_USER_OPTIONS_FILE_PATH="${HOME}/.my.cnf"
...
+    "postgres"
```
Comment (mathieupeterson) [inline ubuntu-18.04/mysql.sh]: postgres?

Comment (PatrickMessierGoC) [reply ubuntu-18.04/mysql.sh:46]: "The postgres database is a default database meant for use by users, utilities and third party applications."
Ref.: https://www.postgresql.org/docs/current/app-initdb.html

Comment (PatrickMessierGoC) [reply ubuntu-18.04/mysql.sh]: Good catch...

File: ubuntu-18.04/mysql.sh (hunk 1 above)
Comment (PatrickMessierGoC) [reply ubuntu-18.04/mysql.sh]: Good catch...

**Review by mathieupeterson — APPROVED** [review]

**Pull request merged**